packages/discovery-react-components/src/components/StructuredQuery/components/AddRuleRowButton/AddRuleRowButton.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { FC, Dispatch, SetStateAction } from 'react';
+import { Button } from 'carbon-components-react';
+import Add16 from '@carbon/icons-react/lib/add/16';
+import { Messages } from 'components/StructuredQuery/messages';
+import { StructuredQuerySelection } from 'components/StructuredQuery/utils/structuredQueryInterfaces';
+
+export interface AddRuleRowButtonProps {
+  /**
+   * text to display for the Add rule button
+   */
+  addRuleRowText: Messages['addRuleRowText'];
+  /**
+   * id of the group for the rule row to render, or 'top-level' if the top-level rule group
+   */
+  groupId: number | 'top-level';
+  /**
+   * state that represents the current rules and selections for the structured query
+   */
+  structuredQuerySelection: StructuredQuerySelection;
+  /**
+   * used to set the structuredQuerySelection state
+   */
+  setStructuredQuerySelection: Dispatch<SetStateAction<StructuredQuerySelection>>;
+}
+
+const handleOnClick = (
+  structuredQuerySelection: StructuredQuerySelection,
+  setStructuredQuerySelection: Dispatch<SetStateAction<StructuredQuerySelection>>,
+  groupId: number | 'top-level'
+) => {
+  if (groupId === 'top-level') {
```

Comment:
Would it cut out any additional code to just check if it's the first group (or id == 0), rather than making the id `number | 'top-level'`, or is it not guaranteed to be first (or have 0 as the id)?

Reply:
Yeah, the top-level group doesn't have an `id` at all as I have it set up, since there can only be one top-level that contains rows and groups. Which is why I ended up using the `top-level` here. If it had an `id`, then would need another way to know that it was top-level and not nested.